src/metadata.mjs

```diff
@@ -1,13 +1,19 @@
 import { cacheGet, cachePut } from "./cache.mjs";
 import { enqueueMetadataFetching } from "./titiler_fetcher.mjs";
 import { keyFromS3Url } from "./key_from_s3_url.mjs";
+import { logger } from "./logging.mjs";
 
 async function cacheGetMetadata(key) {
+  logger.debug(`Got metadata key: ${key}`);
+
   const buffer = await cacheGet(`__metadata__/${key}.json`);
   if (buffer === null) {
     return null;
   }
 
+  logger.debug(`Cached metadata buffer: ${buffer}`);
```

Comment:
for what this if below we write the same info but stringified?

Reply:
to check the buffer var to nullability